src/ejc_sql/structure.clj

```diff
@@ -52,15 +52,19 @@
   "Execute `sql`, return first value of first column of result set as result."
   (first (db->column db sql :row-fn row-fn)))
 
-(defn get-ms-sql-server-version [db]
+(defn get-ms-sql-server-version [ver]
   (->
-   (filter not-empty
-           (.split
-            (db->value db "SELECT @@VERSION AS 'SQL Server Version'")
-            " "))
+   (filter not-empty (.split ver " "))
    (nth 3)
    Integer/parseInt))
 
+(defn is-legacy-sql-server? [db]
+  (let [ver (db->value db "SELECT @@VERSION AS 'SQL Server Version'")]
+    (if (.startsWith ver "Microsoft SQL Azure")
+      false
+      (> (get-ms-sql-server-version ver) 2000))))
```

Comment:
Unfortunately, I don't have any MS SQL Server instances to test but looks like we have an inversed logic here. 
Literally: `if version > 2000` - it's **not** a legacy.
Could you update the Pull Request, please?

Reply:
You're right of course. I switched the name of the function since we're actually testing for the inverse of legacy. This reads better than adding a `(not` to the usages.